theano/sandbox/gpuarray/basic_ops.py

```diff
@@ -259,57 +329,59 @@ def c_code_cache_version(self):
 
     def grad(self, inputs, grads):
         gz, = grads
-        return [gpu_from_host(gz)]
+        return [GpuFromHost(inputs[0].type.context_name)(gz)]
 
     def R_op(self, inputs, eval_points):
         ev, = eval_points
-        if isinstance(ev, tensor.TensorType):
-            return [gpu_from_host(ev)]
-        else:
-            return [ev]
+        return self(ev)
```

Comment:
Why this? If needed, I think we should do the same in the old back-end for now to help keep them in sync.